src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobInstanceService.scala

```diff
@@ -33,4 +39,19 @@ class JobInstanceServiceImpl(override val jobInstanceRepository: JobInstanceRepo
     jobInstanceRepository.getJobInstances(dagInstanceId)
   }
 
+  override def killJob(applicationId: String)(implicit ec: ExecutionContext): Future[Boolean] = {
+    val url: String =
+      s"${SparkExecutorConfig.getHadoopResourceManagerUrlBase}/ws/v1/cluster/apps/$applicationId/state?user.name=${SparkExecutorConfig.getUserUsedToKillJob}"
+    val data = Json.obj(
+      "state" -> "KILLED"
+    )
+
+    Try {
+      WSClientProvider.getWSClient.url(url).put(data).map { response =>
+        response.status == 200 || response.status == 202
+      }.recoverWith {
```

Comment:
Nice, so there's no need to update any status, because the status is updated anyway by the usual scheduler loop. That's really nice and decoupled

Reply:
Exactly